pkg/route/api/validation/validation.go

```diff
@@ -32,6 +32,11 @@ func ValidateRoute(route *routeapi.Route) fielderrors.ValidationErrorList {
 		result = append(result, fielderrors.NewFieldInvalid("path", route.Spec.Path, "path must begin with /"))
 	}
 
+	if len(route.Spec.Path) > 0 && route.Spec.TLS != nil &&
+		route.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough {
```

Comment:
Well, that one's a doozy - can we ever be sure of that? What we don't know of could always exist on a different plane of existence (aka schrodinger's cat). There could well be a script out there that expects things that worked in v1beta1 to work now or in the future and are we absolutely certain that we can ever have a 100% temporal guarantee on that. IMHO, within a reasonable scheme/limit is good. And in this specific case, this is something that would not have worked in the first place. Just me tuppence.